rpcs3/util/sysinfo.cpp

```diff
@@ -358,6 +359,16 @@ std::string utils::get_OS_version()
 	return output;
 }
 
+#ifndef _WIN32
+int utils::get_maxfiles()
+{
+	struct rlimit limits;
+	getrlimit(RLIMIT_NOFILE, &limits);
```

Comment:
Could use `ensure` to make sure this sets a valid value to limits. I guess there's no reason why it would, but at same time I don't think the check would hurt performance either.

You also could add this function to all operating systems and make it return max int for windows (with a comment stating why), that'd get rid of the ifndef mess on other files. But that's just a suggestion.